python/renderer_status.py

```diff
@@ -6,6 +6,8 @@ def main(args):
     # Tell the API the current render sttus
     try:
         # get passed-in arguments
+        args.setdefault('part_id', None)
```

Comment:
This is not a very common Python idiom (in fact I never knew it even existed!). The more common approach is instead of:

```python
        args.setdefault('part_id', None)
        args.setdefault('status', 'new')
        part_id = args['part_id']
        status = args['status']
```

to use:

```python
        part_id = args.get('part_id')
        status = args.get('status', 'new')
```

By default it will return `None` as the default value unless a default given (e.g. `new` above)